src/Numerics/DGmethods/DGmodel.jl

```diff
@@ -79,6 +83,10 @@ function (dg::DGModel)(
     ndrange_interior_surface = Nfp * length(grid.interiorelems)
     ndrange_exterior_surface = Nfp * length(grid.exteriorelems)
 
+    if !increment && num_state_conservative < num_state_tendency
+        tendency .= -zero(FT)
+    end
```

Comment:
What would need to change so that we didn't do this zeroing? Just wondering since this could be a lot of excess memory writes (since we only have 5 or 6 conservative variables and 10s of tracers).

Is it the remainder model and IMEX solvers that don't know about the difference between conservative and tendency?

Though probably not a huge deal since filling an array should be fairly efficient compared to everything else we do in the kernels below.

Reply:
Yea, I think the issue is we are sharing vectors between the models.  We would need to define what it means to add a vector with 3 states to a vector with 10 states.

Reply:
> Yea, I think the issue is we are sharing vectors between the models. We would need to define what it means to add a vector with 3 states to a vector with 10 states.

Can this be done with vars_index / vars-aware MPIStateArrays? Being to add vectors of different sizes but with overlapping variable names could be very useful elsewhere as well

Reply:
> Can this be done with vars_index / vars-aware MPIStateArrays?

Possibly. I would say we should get this in, and then revisit it in a future PR. Maybe open an issue?